### DISCUSSION TITLE
Duplication of Cloze / Double Side

### DISCUSSION THREAD

**Discussion by eMerzh** [h16nning/skola#61] [Q&A] [answered]
Hello 👋 

It seems that "cloze cards" and "double-sided cards" are materialized as multiple cards.
For example: you create a double-sided card, you end up with two cards represented in "manage" view or in "notebook" view.

My guess is that you mapped the "learning" way of looking at the deck (vs the card way, or edit way i would way).

Currently, I touch more around the edition, so maybe that explain the question, 
but the more I touch, the more I'm confused by this. 

The question is "what are the reasons behind this choice ?" because i might see only a tip of the iceberg ...

The reflexion is : 
  Maybe we could keep only 1 card for those cases, and when learning, we can "generate" or map or whatever, the other "sides" or other cards, to make it works

**Comment by h16nning** [comment]
I'm sorry, I don't really understand what you are trying to say 😅😅

**Reply by eMerzh** [reply]
hum shit... i write as thought... sorry it's a mess 😓 

i rephrased it.. hope it's easier?

**Comment by h16nning** [comment]
Oh now I understand.

> My guess is that you mapped the "learning" way of looking at the deck (vs the card way, or edit way i would way).
Exactly. The main reason I decided to have multiple cards because two card sides might be very different. As an example
Say I have a card double-sided vocab card: Hello < - > Hola. I might remember  this way Hola → Hello much better than the opposite way. As such they need a different model object (due date, stability, difficulty).  They might be shown on different days and could "live relatively separate lifes" after being created. They will also have separate statistics and so on.

### Anki
Anki has a bit more complex system where it differentiates between "notes" and "cards". Ankis cards work similar to ours. Notes basically are the common grounds with all their fields from which the individual cards stem, as far as I understand. In their browse menu you can toggle between searching for notes or cards.

<img width="80" alt="Bildschirmfoto 2024-02-14 um 22 41 27" src="https://github.com/h16nning/skola/assets/48356881/bc36d8fd-2348-4ff6-9dfe-08a24bdc7dc0">

I decided against cards and notes as I found this terminology to be hard to grasp. Instead I implemented shared values where the field only references another value. As a context, referencing that values updates all other cards.

### Solution?
I agree with you that it may be a little confusing right now.

I see two options:

We could work around the confusing things. I don't know what confuses you specifically so maybe you can name those things?

I thought about adding a related card sections below the edit box where cards referencing the same shared values are linked.

There is also the issue in NotebookView where some information might be shown multiple times (especially if something like alwaysShowAnswer would be true there).

What else confuses you?

The other option is a rewrite of the logic. However that might require a lot of work. And also we still would need to integrate a different repetition model, statistics etc. for all cards. I don't really know how that could be structured right now.

Other than that, I actually like the way shared values work right now. Maybe some clearer UI and checks to hide "duplicate" information would be enough?

**Reply by eMerzh** [reply]
Thanks for the detailed response 🙌 

first: it was like 5 years i didn't open anki, i did to better understand what you said... omg... it's a mess 😱  i don't understand anything 😓 

To be more specific, i don't get why you would want to see multiple card ever... 
You add that card once, it should be displayed once.
The fact that you'll learn it one side then other should be hidden.
I DO get that you want a way to store the stats per "sub-card" or something.

I was confused to see multiple time the same card until I realized it was because of double-sided (the issue is more visible since it's hard to grasp at first glance that the card is "special type")

Then same thing for cloze cards, i couldn't understand why the system understood only One of my cloze field... because it's one per card... of course...  

Also you've added the custom sort (which is ok of course) but then the cards are somewhat all over the place by default
one cloze at 3rd on 20 the other 14th on 20 and the other 18th on 20 🤨 .  (i know by *definition* i need to order them)


then there's the technical side, i played with an cmd+k search implementation.
But the results were quickly cluttered with a lot of "duplicated" card... 
We should probably dedup them also in notebook view for example.
Then i was looking to move a card to another deck, you have to maintain all the cards everytime ... 

It feels like you do the heavy work "often" to simplify only "1 screen" if you understand what I mean.

Note: I don't say we need to rewrite everything NOW or blame or 🤷, I try to understand, and i respect your choices.
Just here to discuss the best path to make the app better :)

**Reply by h16nning** [reply]
I totally get your point.

I like the association: one card = one piece of information with it is information of how well i can remember it etc.

The question is how do we fix it.

We could do a rewrite with "sub-cards" (although I find that it "sub-cards" sounds a little confusing) but it sounds like a lot of work...

One quick and dirty fix would be to create something like a index / head card of related cards and only show that and hide the other ones.

**Reply by eMerzh** [reply]
i have NO idea 😅  i was not going to propose a rewrite here 😬 i promise... 
i try to give it some thoughts.... but i still need to figure out how dexie works and what you can do with it

**Comment by h16nning** [comment]
A cmd+k search sounds very cool btw. What did you think about, something like a spotlight search?

**Reply by eMerzh** [reply]
i can publish the experiment ... it's crappy ... but so you can see :p

**Reply by eMerzh** [reply, marked as answer]
#63
